zson/analyzer.go

```diff
@@ -595,10 +595,11 @@ func (a Analyzer) convertType(zctx *Context, typ ast.Type) (zng.Type, error) {
 	case *ast.TypeName:
 		typ, ok := a[t.Name]
 		if !ok {
-			typ = zctx.LookupTypeDef(t.Name)
-			if typ == nil {
+			alias := zctx.LookupTypeDef(t.Name)
+			if alias == nil {
 				return nil, fmt.Errorf("no such type name: %q", t.Name)
 			}
+			typ = alias
```

Comment:
Nit: I think the original code was clearer.

Reply:
It was clearer but it had the interface-nil bug because typ is a zng.Type and LookupTypeDef returns *zng.TypeAlias.  I will add a comment in the code.  (I had put a note in the PR comment.)